workingspacebooking/src/main/java/naitei/group5/workingspacebooking/service/impl/AccountRecoveryServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+import naitei.group5.workingspacebooking.entity.AccountRecovery;
+import naitei.group5.workingspacebooking.entity.User;
+import naitei.group5.workingspacebooking.entity.enums.AccountRecoveryStatus;
+import naitei.group5.workingspacebooking.repository.AccountRecoveryRepository;
+import naitei.group5.workingspacebooking.repository.UserRepository;
+import naitei.group5.workingspacebooking.service.AccountRecoveryService;
+import naitei.group5.workingspacebooking.service.EmailService;
+import naitei.group5.workingspacebooking.dto.request.PasswordResetRequest;
+import naitei.group5.workingspacebooking.utils.TokenGenerator;
+import static naitei.group5.workingspacebooking.constant.Endpoint.AUTH_RECOVERY_CONFIRM;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AccountRecoveryServiceImpl implements AccountRecoveryService {
+
+    private final UserRepository userRepository;
+    private final AccountRecoveryRepository accountRecoveryRepository;
+    private final EmailService emailService;
+    private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+    @Override
+    public void initiateRecovery(String email) {
+        Optional<User> optionalUser = userRepository.findByEmail(email);
+        if (optionalUser.isEmpty()) {
+            throw new RuntimeException("Email not found");
+        }
+
+        User user = optionalUser.get();
+
+        // dùng TokenGenerator
+        String token = TokenGenerator.generateToken(30);
+
+        AccountRecovery recovery = AccountRecovery.builder()
+                .user(user)
+                .recoveryToken(token)
+                .expiryDate(LocalDateTime.now().plusMinutes(15))
+                .status(AccountRecoveryStatus.pending)
+                .build();
+        accountRecoveryRepository.save(recovery);
+
+        String resetLink = "http://localhost:8080" + AUTH_RECOVERY_CONFIRM + token;
+
+        emailService.sendSimpleMessage(
+                user.getEmail(),
+                "Password Recovery",
+                "Click this link to reset your password: " + resetLink
+        );
+    }
+
+    @Override
+    public void confirmRecovery(String token, PasswordResetRequest request) {
+        AccountRecovery recovery = accountRecoveryRepository.findByRecoveryToken(token)
+                .orElseThrow(() -> new RuntimeException("Invalid token"));
+
+        if (recovery.getExpiryDate().isBefore(LocalDateTime.now())) {
+            recovery.setStatus(AccountRecoveryStatus.expired);
+            accountRecoveryRepository.save(recovery);
+            throw new RuntimeException("Token expired");
+        }
+
+        User user = recovery.getUser();
+        user.setPassword(passwordEncoder.encode(request.getNewPassword()));
+        userRepository.save(user);
+
+        recovery.setStatus(AccountRecoveryStatus.used);
+        accountRecoveryRepository.save(recovery);
+    }
```

Comment:
1. thêm Transaction để đảm bảo toàn vẹn dữ liệu

2. Áp dụng Optional với phương thức orElseThrow dạng lambda.

3. Sử dụng Exception cụ thể thay vì RuntimeException.